ccl/importccl: TestImportCockroachDump failed

[(ccl/importccl).TestImportCockroachDump failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2174505&tab=buildLog) on [master@ed56390d4f28f7d7b26ce047bcb799afb337b537](https://github.com/cockroachdb/cockroach/commits/ed56390d4f28f7d7b26ce047bcb799afb337b537):

```
=== RUN   TestImportCockroachDump
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestImportCockroachDump849507399
--- FAIL: TestImportCockroachDump (37.92s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestImportCockroachDump849507399
    test_log_scope.go:63: use -show-logs to present logs inline
    import_stmt_test.go:4623: error executing 'IMPORT PGDUMP ($1)': pq: internal error: creating importTables: restoring table desc and namespace entries: validate table 54: missing fk forward reference "fk_i_ref_t" to "t" from "t"

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestImportCockroachDump PKG=./pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestImportCockroachDump.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/ccl/importccl/import_stmt.go
+++ b/pkg/ccl/importccl/import_stmt.go
@@ -49,6 +49,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/humanizeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
@@ -1109,7 +1110,10 @@ func (r *importResumer) prepareTableDescsForIngestion(
 					hasExistingTables = true
 				} else {
 					newTablenameToIdx[table.Desc.Name] = i
-					newTableDescs = append(newTableDescs, table)
+					// Make a deep copy of the table descriptor so that rewrites do not
+					// partially clobber the descriptor stored in details.
+					newTableDescs = append(newTableDescs,
+						*protoutil.Clone(&table).(*jobspb.ImportDetails_Table))
 				}
 			}
 

